File: vital-inventories/src/main/java/me/vitalframework/inventories/VitalInventoryService.java
package me.vitalframework.inventories;

import lombok.NonNull;
import me.vitalframework.Vital;
import org.bukkit.entity.Player;
import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.List;

/**
 * The main vital inventory service for registering inventories.
 */
@Service
public class VitalInventoryService {
    /**
     * Opens a registered {@link VitalInventory} for the given {@link Player}.
     */
    public void openInventory(@NonNull Player player, @NonNull Class<? extends VitalInventory> vitalInventoryClass) {
        final var vitalInventory = Vital.INSTANCE.getContext().getBean(vitalInventoryClass);

        vitalInventory.open(player);
    }

    @NonNull
    public Collection<? extends VitalInventory> getInventories(@NonNull Class<? extends VitalInventory> vitalInventoryClass) {
        try {
            return Vital.INSTANCE.getContext().getBeansOfType(vitalInventoryClass).values();
        } catch (Exception e) {
            return List.of();
        }
    }

    @NonNull
    public Collection<? extends VitalInventory> getInventories() {
        return getInventories(VitalInventory.class);
    }


    public <T extends VitalInventory> T getInventory(@NonNull Class<T> vitalInventoryClass) {
        try {
            return Vital.INSTANCE.getContext().getBean(vitalInventoryClass);
        } catch (Exception e) {
            return null;
        }
    }

    public void updateInventories() {
        getInventories().forEach(VitalInventory::update);
    }
}

File: vital-inventories/src/main/kotlin/me/vitalframework/inventories/VitalInventoryService.kt
package me.vitalframework.inventories

import me.vitalframework.SpigotPlayer
import me.vitalframework.Vital
import org.springframework.stereotype.Service

@Service
class VitalInventoryService {
    fun openInventory(player: SpigotPlayer, vitalInventoryClass: Class<VitalInventory>) {
        val vitalInventory: VitalInventory = Vital.context.getBean(vitalInventoryClass)

        vitalInventory.open(player)
    }

    fun getInventories(vitalInventoryClass: Class<VitalInventory>) = try {
        Vital.context.getBeansOfType(vitalInventoryClass).values.toList()
    } catch (e: Exception) {
        emptyList()
    }

    fun getInventories() = getInventories(VitalInventory::class.java)

    fun <T : VitalInventory> getInventory(vitalInventoryClass: Class<T>) = try {
        Vital.context.getBean(vitalInventoryClass)
    } catch (e: Exception) {
        null
    }

    fun updateInventories() {
        for (inventory in getInventories()) {
            inventory.update()
        }
    }
}